webapp/channels/src/packages/mattermost-redux/src/reducers/entities/users.ts

```diff
@@ -485,29 +485,11 @@ function profilesNotInGroup(state: UsersState['profilesNotInGroup'] = {}, action
     }
 }
 
-function addToState<T>(state: Record<string, T>, key: string, value: T): Record<string, T> {
-    if (state[key] === value) {
-        return state;
-    }
-
-    return {
-        ...state,
-        [key]: value,
-    };
-}
-
-function dndEndTimes(state: RelationOneToOne<UserProfile, number> = {}, action: AnyAction) {
+function dndEndTimes(state: UsersState['dndEndTimes'] = {}, action: AnyAction) {
     switch (action.type) {
-    case UserTypes.RECEIVED_STATUS: {
-        const userId = action.data.user_id;
-        const endTime = action.data.dnd_end_time;
-
-        return addToState(state, userId, endTime);
-    }
-    case UserTypes.RECEIVED_STATUSES: {
-        const userStatuses: UserStatus[] = action.data;
-
-        return userStatuses.reduce((nextState, userStatus) => addToState(nextState, userStatus.user_id, userStatus.dnd_end_time || 0), state);
+    case UserTypes.RECEIVED_DND_END_TIME:
```

Comment:
Regardless of if we go back to the one action or multiple actions, I wonder if we even need the singular actions for these. Unless the "received multiple" action does something different like overwriting all of the items in a list, we I feel like we can usually just dispatch a single object using the "received multiple" action

Reply:
Yeah I agree with Harrison on this, 1/5 on whether you want to change it

Reply:
Yes changed to single reducer instead of two for singular and plural